db/migrate/20220201160946_create_contacts.rb

```diff
@@ -0,0 +1,12 @@
+class CreateContacts < ActiveRecord::Migration[6.1]
+  def change
+    create_table :contacts do |t|
+      t.string :name
+      t.date :bith_date
+      t.references :user, null: false, foreign_key: true
+
+      t.timestamps
+    end
+    add_index :contacts, :name, unique: true
```

Comment:
Essa `index` vai fazer com que a validação falhe por conta da regra de poder repetir os nomes, você pode adicionar um index com ambos os campos para garantir os a integridade no banco

```suggestion
    add_index :contacts, %i[name user_id], unique: true
```